ipaserver/install/ipa_cert_fix.py

```diff
@@ -0,0 +1,244 @@
+#
+# Copyright (C) 2019  FreeIPA Contributors see COPYING for license
+#
+
+# ipa-cert-fix performs the following steps:
+#
+# 1. Confirm running as root (AdminTool.validate_options does this)
+#
+# 2. Confirm that DS is up.
+#
+# 3. Determine which of following certs (if any) need renewing
+#     - IPA RA
+#     - Apache HTTPS
+#     - 389 LDAPS
+#     - Kerberos KDC (PKINIT)
+#
+# 4. Execute `pki-server cert-fix` with relevant options,
+#    including `--extra-cert SERIAL` for each cert from #3.
+#
+# 5. Print details of renewed certificates.
+#
+# 6. Install renewed certs from #3 in relevant places
+#
+# 7. ipactl restart
+
+from __future__ import print_function, absolute_import
+
+import datetime
+from enum import Enum
+import logging
+import shutil
+import subprocess
```

Comment:
Please use ``ipapython.ipautil.run`` instead of subprocess. Our helper does additional logging.